pylint/testutils/run.py

```diff
@@ -0,0 +1,46 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+from __future__ import annotations
+
+from collections.abc import Sequence
+from pathlib import Path
+from typing import Any
+
+from pylint.lint import Run as LintRun
+from pylint.lint.run import UNUSED_PARAM_SENTINEL
+from pylint.reporters.base_reporter import BaseReporter
+from pylint.testutils.lint_module_test import PYLINTRC
+
+
+def add_rcfile_pylintrc_to_args(pylintrc: str | Path, args: list[str]) -> list[str]:
+    """Add the rcfile option to the pylintrc path to an args list."""
+    if not any("--rcfile" in arg for arg in args):
+        args.insert(0, f"--rcfile={pylintrc}")
+    return args
+
+
+def add_rcfile_default_pylintrc(args: Sequence[str]) -> list[str]:
+    """Add a default pylintrc with the rcfile option in a list of pylint args."""
+    return add_rcfile_pylintrc_to_args(PYLINTRC, list(args))
+
+
+class Run(LintRun):
```

Comment:
You have probably explored this, but can't we just pass an empty `rcfile` whenever we want to? Overwriting `Run` feels like an overcomplicated solution. We already support ``rcfile=``, why not use that?

_This just adds more stuff to maintain._

Reply:
Yeah I started by passing rcfile "when needed" but after adding an init-hook to make a crash if we were using the project's pylintrc there was hundreds of failing tests. May as well modify the class that is being called by the test (in this case only the import from pylint.testutils instead of pylint.lint). Adding a new argument as a string for each tests is a lot of work (litteral hundred of tests to fix). Note that sometime I use the internal function directly.

Reply:
Hmm shall we make it private then? We need to deprecate one of the parameters to `Run` in `3.0` and having to do so in another class makes that more complicated. Since we don't really have anybody asking for us to add this to `testutils` we don't need to make it public.

Reply:
194e84215